invenio_rdm_records/requests/access/requests.py

```diff
@@ -0,0 +1,215 @@
+
+    def __init__(self, receiver, subject, html_body, body):
+        """Initialize operation."""
+        self.receiver = receiver
+        self.subject = subject
+        self.html_body = html_body
+        self.body = body
+
+    def on_post_commit(self, uow):
+        """Start task to send text via email."""
+        send_email(
+            {
+                "subject": self.subject,
+                "html_body": self.html_body,
+                "body": self.body,
+                "recipients": [self.receiver],
+                "sender": current_app.config["MAIL_DEFAULT_SENDER"],
+            }
+        )
+
+
+#
+# Actions
+#
+class UserSubmitAction(actions.SubmitAction):
+    """Submit action for user access requests."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        record = self.request.topic.resolve()
+        creator = self.request.created_by.resolve()
+
+        title = record.metadata["title"]
+        name = creator.username or creator.email.split("@")[0]
+        if creator.user_profile:
+            name = creator.user_profile["full_name"]
+
+        self.request["title"] = f"{name} requests access to {title}"
+        super().execute(identity, uow)
+
+
+class GuestSubmitAction(actions.SubmitAction):
+    """Submit action for guest access requests."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        record = self.request.topic.resolve()
+        payload = self.request["payload"]
+        name = f"{payload['full_name']} ({payload['email']})"
+        title = record.metadata["title"]
+
+        self.request["title"] = f"{name} requests access to {title}"
+        super().execute(identity, uow)
+
+
+class GuestAcceptAction(actions.AcceptAction):
+    """Accept action."""
+
+    def execute(self, identity, uow):
+        """Accept record into community."""
+        record = service.read(
+            id_=self.request.topic.reference_dict["record"], identity=system_identity
+        )
+        payload = self.request["payload"]
+
+        data = {
+            # "expires_at": None,  # TODO enable expiration
+            "permission": payload["permission"],
+            "description": f"Requested by guest: {payload['full_name']} ({payload['email']})",
+            "origin": f"request:{self.request.id}",
+        }
+
+        link = service.access.create_secret_link(identity, record.id, data)
+        access_url = f"{record.links['self_html']}?token={link._link.token}"
+        plain_message = f"Access the record here: {access_url}"
+        message = f'Click <a href="{access_url}">here</a> to access the record.'
+        comment_data = {"payload": {"content": message}}
+        events_service.create(
+            system_identity, self.request.id, comment_data, CommentEventType, uow=uow
+        )
+
+        uow.register(RecordCommitOp(record._record.parent))
+        uow.register(
+            EmailOp(
+                receiver=payload["email"],
```

Comment:
Question: Not for now, would it make sense make the emails easily customizable? via a jinja template for instance

Reply:
Definitely!

Reply:
If a notification is used, this would be possible out of the box (same goes for recipients and subject).